### PR TITLE
Stellar: sign user provided transaction envelope

### PR DESCRIPTION
**Pull request by zapu**
Add a command and an RPC to sign arbitrary transactions.

### CODE REVIEW

**Comment by zapu** [comment]
Added a test to see if the call reaches (non-mocked) remote handler.

File: go/client/cmd_wallet_sign.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+func newCmdWalletSign(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	cmd := &CmdWalletSign{
+		Contextified: libkb.NewContextified(g),
+	}
+	flags := []cli.Flag{
+		cli.StringFlag{
+			Name:  "xdr",
+			Usage: "Transaction envelope XDR. If not provided, will be read from stdin.",
```
Comment (patrickxb) [inline go/client/cmd_wallet_sign.go]: Could you add that it is base64-encoded?

File: go/stellar/stellarsvc/frontend.go [hunk 2]
```diff
@@ -1201,3 +1202,73 @@ func (s *Server) FindPaymentPathLocal(ctx context.Context, arg stellar1.FindPaym
...
+		// Derive signer account id from transaction's sourceAccount.
+		accountID = stellar1.AccountID(unpackedTx.Tx.SourceAccount.Address())
+		mctx.Debug("Trying to sign with SourceAccount: %s", accountID.String())
+	} else {
+		// We were provided with specific AccountID we want to sign with.
+		accountID = *arg.AccountID
+		mctx.Debug("Trying to sign with (passed as argument): %s", accountID.String())
+	}
+
+	_, accBundle, err := stellar.LookupSender(mctx, accountID)
```
Comment (patrickxb) [inline go/stellar/stellarsvc/frontend.go]: can you change `accBundle` to `acctBundle`

File: go/stellar/stellarsvc/frontend.go [hunk 3]
```diff
@@ -1201,3 +1202,73 @@ func (s *Server) FindPaymentPathLocal(ctx context.Context, arg stellar1.FindPaym
 
 	return res, nil
 }
+
```
Comment (patrickxb) [inline go/stellar/stellarsvc/frontend.go]: could you move these to stellarsvc/service.go since they are CLI endpoints?

**Pull request merged**